### DISCUSSION TITLE
Can I use Amazon Bedrock for Langfuse Evals?

### DISCUSSION THREAD

**Discussion by j10sanders** [langfuse/langfuse#3698] [Support] [answered]
### Describe the feature or potential improvement

I was excited to start using your platform, but am blocked because my company can't send data to openai. We use Amazon Bedrock for compliance reasons.

When I go to create an eval template, I see `No LLM API key set in project. For evals, only OpenAI models are supported.`

Please consider adding other providers - specifically Bedrock for my use case.

**Comment by j10sanders** [comment]
Is it possible to do this in the langfuse dashboard while using uptrain, which sorta has bedrock support?

https://github.com/uptrain-ai/uptrain/issues/589

**Comment by marcklingen** [comment]
cc @hassiebp who is currently working on extending support for bedrock

**Reply by j10sanders** [reply]
wow that would be amazing - especially if possible by end of year! 🙏

**Comment by hassiebp** [comment, marked as answer]
@j10sanders - we have just shipped this: https://langfuse.com/changelog/2024-10-11-bedrock-support-playground-evals

Happy to hear your feedback! 🙏🏾

**Comment by j10sanders** [comment]
At @hassiebp that's awesome! Thank you

I'm currently unable to set this up. Getting an error in the console: `Error: User: arn:aws:iam::219042630441:user/Engineering/JonathanSanders is not authorized to perform: bedrock:InvokeModel on resource: arn:aws:bedrock:us-west-2::foundation-model/anthropic.claude-3-sonnet-20240229-v1:0 with an explicit deny in an identity-based policy`

Can you let me know the exact Bedrock IAM authorizations you need?

https://github.com/user-attachments/assets/51a16a5a-9099-4beb-8f3e-89b8f755bdc0

**Reply by hassiebp** [reply]
Does your user have permissions for `bedrock:InvokeModel` the action?

**Reply by j10sanders** [reply]
Just want to check if that is the only permission I need here

**Reply by hassiebp** [reply]
`InvokeModel` should be sufficient. I have tested with the AWS managed `AmazonBedrockFullAccess` policy which worked 👍🏾

**Comment by j10sanders** [comment]
Great that did work! Thank you @hassiebp 

I do have a suggestion:

We would prefer if you weren't asking for access keys/secrets. Instead, we'd like to be able to create a role that can be assumed your AWS account. We would authorize the AWS account and provide you with the ARN of the AWS role to assume.

In the way it is currently implemented, I would need to create another AWS user in order to have more than 2 models used (or even for more than 1 model across 2 projects) because only 2 access keys can be assigned to a user.

**Reply by marcklingen** [reply]
Awesome!

> We would prefer if you weren't asking for access keys/secrets. Instead, we'd like to be able to create a role that can be assumed your AWS account. We would authorize the AWS account and provide you with the ARN of the AWS role to assume.

Thanks for the feedback, can you create a new thread for this?

> In the way it is currently implemented, I would need to create another AWS user in order to have more than 2 models used (or even for more than 1 model across 2 projects) because only 2 access keys can be assigned to a user.

You should be able to reference more than 2 models via the connection. Can you share what the limiting factor here is?

**Reply by j10sanders** [reply]
Sure I'll make a new thread for the first part.

For the second - hmm, I guess I could use the same key id and access key across multiple models. Ok nevermind!
